app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/QuestionDualImageViewModel.kt

```diff
@@ -0,0 +1,88 @@
+package org.kinecosystem.kinit.viewmodel.earn
+
+import org.kinecosystem.kinit.KinitApplication
+import org.kinecosystem.kinit.analytics.Analytics
+import org.kinecosystem.kinit.analytics.Events
+import org.kinecosystem.kinit.model.earn.Answer
+import org.kinecosystem.kinit.model.earn.Question
+import org.kinecosystem.kinit.model.earn.Task
+import org.kinecosystem.kinit.model.earn.tagsString
+import org.kinecosystem.kinit.repository.TasksRepository
+import org.kinecosystem.kinit.util.Scheduler
+import org.kinecosystem.kinit.view.customView.AnswerSelectedOverView.OnSelectionListener
+import org.kinecosystem.kinit.view.earn.QuestionnaireActions
+import javax.inject.Inject
+
+class QuestionDualImageViewModel(private var questionIndex: Int,
+                                 private val questionnaireActions: QuestionnaireActions?) {
+    @Inject
+    lateinit var scheduler: Scheduler
+    @Inject
+    lateinit var questionnaireRepository: TasksRepository
+    @Inject
+    lateinit var analytics: Analytics
+
+    private val POST_ANSWER_DELAY: Long = 400
+    private var question: Question? = null
+    var chosenAnswers: MutableList<String> = mutableListOf()
+    var imageUrls: MutableList<String> = mutableListOf()
+    var selectionListeners: MutableList<OnSelectionListener> = mutableListOf()
```

Comment:
would be simpler to have a single selectionListener that takes the index of the answer as a parameter

Reply:
how would you send the index of and answer